packages/shylu/shylu_dd/frosch/src/CoarseSpaces/FROSch_LocalPartitionOfUnityBasis_def.hpp

```diff
@@ -142,7 +142,7 @@ namespace FROSch {
                     for (UN j=0; j<maxNV[i]; j++) {
                         //MultiVectorPtrVecPtr tmpBasis2(PartitionOfUnity_[i]->getNumVectors());
                         MultiVectorPtr entityBasis = Xpetra::MultiVectorFactory<SC,LO,GO,NO >::Build(PartitionOfUnity_[i]->getMap(),PartitionOfUnity_[i]->getNumVectors());
-                        entityBasis->scale(0.0);
+                        entityBasis->scale(Teuchos::ScalarTraits<SC>::zero());
                         for (UN k=0; k<PartitionOfUnity_[i]->getNumVectors(); k++) {
                             if (j<tmpBasis[i][k]->getNumVectors()) {
                                 entityBasis->getDataNonConst(k).deepCopy(tmpBasis[i][k]->getData(j)()); // Here, we copy data. Do we need to do this?
```

Comment:
What are you trying to do here?  It looks like you are trying to make a deep copy of a MultiVector.  We have functions for doing that.

Reply:
@mhoemmen I am not only copying but also permuting rows.

Reply:
@searhein Do you have Maps that represent the permutation?  If so, you could just rely on doExport / doImport to do the permutation.

Reply:
@mhoemmen I am sorry. I meant that I am permuting columns.